src/main/java/com/github/checkstyle/patchfilter/SuppressionPatchFilter.java

```diff
@@ -92,25 +97,69 @@ protected void finishLocalSetup() throws Exception {
     }
 
     private PatchFilterSet loadPatchFile(String patchFileName) throws IOException {
-        final List<String> patched = Files.readAllLines(new File(patchFileName).toPath());
-        final String fileName = getFileName(patched);
-        final Patch<String> patch = UnifiedDiffUtils.parseUnifiedDiff(patched);
-        final Chunk chunk = patch.getDeltas().get(0).getTarget();
-        final Integer startLine = chunk.getPosition();
-        final Integer endLine = chunk.getPosition() + chunk.getLines().size();
-        final SuppressionPatchFilterElement element =
-                new SuppressionPatchFilterElement(fileName, startLine, endLine);
-        filters.addFilter(element);
+        final List<String> originPatch = Files.readAllLines(new File(patchFileName).toPath());
+        final List<List<String>> patchList = getPatchList(originPatch);
+        for (List<String> singlePatch : patchList) {
+            final String fileName = getFileName(singlePatch);
+            final Patch<String> patch = UnifiedDiffUtils.parseUnifiedDiff(singlePatch);
+            final List<List<Integer>> lineRangeList = getLineRange(patch);
+            final SuppressionPatchFilterElement element =
+                    new SuppressionPatchFilterElement(fileName, lineRangeList);
+            filters.addFilter(element);
+        }
         return filters;
     }
 
     /**
-     * To get file name that is changed.
-     * @param patched patch file's List of string
-     * @return String
+     * To seperate different files's diff contents when there are multiple files in patch file.
+     * @param originPatch List of String
+     * @return patchedList List of List of String
      */
-    private String getFileName(List<String> patched) {
-        return patched.get(1).split("\\s")[1];
+    private List<List<String>> getPatchList(List<String> originPatch) {
+        final List<List<String>> patchedList = new ArrayList<>();
+        boolean flag = true;
+        List<String> singlePatched = new ArrayList<>();
+        for (String str : originPatch) {
+            if (str.startsWith("diff -u ")) {
```

Comment:
It is not a part of a diff. Why do you use this information to split the files?

Reply:
when using diff to compare two directories, this line will be part of a patch

Reply:
Please give example, I don't see this line is a part of a patch:
```
/var/tmp/checkstyle $ ls a b
a:
Origin.java
b:
Update.java

/var/tmp/checkstyle $ diff -u a/Origin.java b/Update.java 
--- a/Origin.java	2020-06-12 08:34:17.002841681 +0100
+++ b/Update.java	2020-06-12 08:34:26.576808407 +0100
@@ -6,6 +6,11 @@
         return;
       } else if (error > 40) {
         System.out.println("ok");
+        return;
+      } else if (error > 30) {
+        return;
+      } else {
+        System.out.println("pass");
       }
     }
 }
```

Reply:
This need more than one file in every directories.
Origin dir:

cat Origin/test1.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Origin;

public class Test1 {
    public void test1() {

    }
}

```
cat Origin/test2.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Origin;

public class Test2 {
    public void test1() {
        System.out.println();
    }
}
```
Update dir:

Update/test1.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Update;

public class Test1 {
    public void test1() {

    }

    public void test2() {

    }

    public void test3() {

    }
}

```
Update/test2.java
```java
package checkstyle_demo.PatchSuppression.multiFile.Update;

public class Test2 {
    public void test1() {
        System.out.println();
        System.out.println();
        System.out.println();
    }
}
```
run diff command 

```sh
diff -u Origin/ Update/ > multiPatch.txt
```
cat multiPatch.txt
```
diff -u Origin/Test1.java Update/Test1.java
--- Origin/Test1.java	2020-06-11 10:48:40.000000000 +0800
+++ Update/Test1.java	2020-06-11 10:48:40.000000000 +0800
@@ -1,7 +1,15 @@
-package checkstyle_demo.PatchSuppression.multiFile.Origin;
+package checkstyle_demo.PatchSuppression.multiFile.Update;
 
 public class Test1 {
     public void test1() {
 
     }
+
+    public void test2() {
+
+    }
+
+    public void test3() {
+
+    }
 }
diff -u Origin/Test2.java Update/Test2.java
--- Origin/Test2.java	2020-06-11 10:48:40.000000000 +0800
+++ Update/Test2.java	2020-06-11 10:48:41.000000000 +0800
@@ -1,7 +1,9 @@
-package checkstyle_demo.PatchSuppression.multiFile.Origin;
+package checkstyle_demo.PatchSuppression.multiFile.Update;
 
 public class Test2 {
     public void test1() {
         System.out.println();
+        System.out.println();
+        System.out.println();
     }
 }

```

Reply:
This is a nuances of diff tool that it prints command line for each file, it is not a case for patch files. Please open any git diff  to see hw it looks. https://github.blog/2011-10-21-github-secrets/#diff-patch

Reply:
there is `diff --git` above each file - https://github.com/checkstyle/checkstyle/commit/c42a0e1b63c33400e05c98b62179cf303724e525.patch

```
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
index adae0fe2d0..46bd2bf257 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java
```

Reply:
done